From b79e40f57336f67008963796a8987d320d0d5d51 Mon Sep 17 00:00:00 2001
From: Volker Hilsheimer <volker.hilsheimer@qt.io>
Date: Mon, 15 Sep 2025 15:49:38 +0200
Subject: [PATCH 093/553] Add security tags to src/gui/itemmodels

QFileSystemModel parses and processes file paths, which includes complex
logic for e.g. following symlinks. Mistakes here could easily introduce
vulnerabilities. In addition, corrupted file paths might result in
undefined behavior in the parsing code. Mark the implementation of that
class (including the private header) as security critical. The public
header only contains the declaration of a more or less typical
QAbstractItemModel subclass, and can stick to the default.

The remaining code in this folder can also  stay with the default
classification of 'significant'.

QUIP: 23
Fixes: QTBUG-135218
Pick-to: 6.9 6.8
Change-Id: Iaefb5ddb02a061d67236759efc23c7ced533f361
Reviewed-by: Mate Barany <mate.barany@qt.io>
(cherry picked from commit d95bc3bf9aa44b2ed9e5bee9699da8e53c319455)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/gui/itemmodels/qfileinfogatherer.cpp  | 1 +
 src/gui/itemmodels/qfileinfogatherer_p.h  | 1 +
 src/gui/itemmodels/qfilesystemmodel.cpp   | 1 +
 src/gui/itemmodels/qfilesystemmodel.h     | 1 +
 src/gui/itemmodels/qfilesystemmodel_p.h   | 1 +
 src/gui/itemmodels/qstandarditemmodel.cpp | 1 +
 src/gui/itemmodels/qstandarditemmodel.h   | 1 +
 src/gui/itemmodels/qstandarditemmodel_p.h | 1 +
 8 files changed, 8 insertions(+)

diff --git a/src/gui/itemmodels/qfileinfogatherer.cpp b/src/gui/itemmodels/qfileinfogatherer.cpp
index df43f53050a..a752f12b002 100644
--- a/src/gui/itemmodels/qfileinfogatherer.cpp
+++ b/src/gui/itemmodels/qfileinfogatherer.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qfileinfogatherer_p.h"
 #include <qcoreapplication.h>
diff --git a/src/gui/itemmodels/qfileinfogatherer_p.h b/src/gui/itemmodels/qfileinfogatherer_p.h
index 5dbda50d94a..ee14cd864c2 100644
--- a/src/gui/itemmodels/qfileinfogatherer_p.h
+++ b/src/gui/itemmodels/qfileinfogatherer_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QFILEINFOGATHERER_H
 #define QFILEINFOGATHERER_H
diff --git a/src/gui/itemmodels/qfilesystemmodel.cpp b/src/gui/itemmodels/qfilesystemmodel.cpp
index 8cfd0650e8f..622c2e5bc92 100644
--- a/src/gui/itemmodels/qfilesystemmodel.cpp
+++ b/src/gui/itemmodels/qfilesystemmodel.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2020 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #include "qfilesystemmodel_p.h"
 #include "qfilesystemmodel.h"
diff --git a/src/gui/itemmodels/qfilesystemmodel.h b/src/gui/itemmodels/qfilesystemmodel.h
index 56aad925561..07970ee5006 100644
--- a/src/gui/itemmodels/qfilesystemmodel.h
+++ b/src/gui/itemmodels/qfilesystemmodel.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QFILESYSTEMMODEL_H
 #define QFILESYSTEMMODEL_H
diff --git a/src/gui/itemmodels/qfilesystemmodel_p.h b/src/gui/itemmodels/qfilesystemmodel_p.h
index e01b0d56e62..0747b2a07d6 100644
--- a/src/gui/itemmodels/qfilesystemmodel_p.h
+++ b/src/gui/itemmodels/qfilesystemmodel_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:critical reason:data-parser
 
 #ifndef QFILESYSTEMMODEL_P_H
 #define QFILESYSTEMMODEL_P_H
diff --git a/src/gui/itemmodels/qstandarditemmodel.cpp b/src/gui/itemmodels/qstandarditemmodel.cpp
index 6c9943d56a8..e26ff1574c0 100644
--- a/src/gui/itemmodels/qstandarditemmodel.cpp
+++ b/src/gui/itemmodels/qstandarditemmodel.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qstandarditemmodel.h"
 
diff --git a/src/gui/itemmodels/qstandarditemmodel.h b/src/gui/itemmodels/qstandarditemmodel.h
index a5ae05feeb1..3eeffdd6c7f 100644
--- a/src/gui/itemmodels/qstandarditemmodel.h
+++ b/src/gui/itemmodels/qstandarditemmodel.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QSTANDARDITEMMODEL_H
 #define QSTANDARDITEMMODEL_H
diff --git a/src/gui/itemmodels/qstandarditemmodel_p.h b/src/gui/itemmodels/qstandarditemmodel_p.h
index 443aafa859a..1eb0d09acba 100644
--- a/src/gui/itemmodels/qstandarditemmodel_p.h
+++ b/src/gui/itemmodels/qstandarditemmodel_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QSTANDARDITEMMODEL_P_H
 #define QSTANDARDITEMMODEL_P_H
-- 
2.51.2

